staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -481,17 +559,21 @@ func structToUnstructured(sv, dv reflect.Value) error {
 		fieldInfo := fieldInfoFromField(st, i)
 		fv := sv.Field(i)
 
+		if !fv.IsValid() {
+			// No fource field, skip.
+			continue
+		}
+		if fieldInfo.omitempty && isZero(fv) {
```

Comment:
@smarterclayton - this is the key optimization that we were missing previously (and that is widely used in our objects).